## Code and data associated with Adkins et al.

### Code:

`phylogeny.R`: script to generate phylogeny figures

### Data

`Cc_Cm_CV_FL_HI_allele_counts_by_species_locality_zeroes_ones.csv`: dataframe of alleles and what population or location they are in, by species; for input to phylogeny_CV_FL_HI.R

### Files

`classI_Hawaiian_Cm_trimmedoutgroup.fasta`: edited Clustal Omega alignment of MHC alleles

`classI_Hawaiian_Cm_trimmedoutgroup.nex`: input nexus file for MrBayes

`infile.nex.con.tre`: MrBayes output consensus tree and input for phylogeny_CV_FL_HI.R

`classI_Hawaiian_Cm_159bp.fasta`: Clustal omega alignment in correct open reading frame and shortened to 159bp, for HyPhy analyses; direct input for GARD analysis

`classI_Hawaiian_Cm_159bp.nex`: input nexus file for MrBayes, for HyPhy analyses

`infile.nex.con_159bp.tre`: output consensus tree of MrBayes, for HyPhy analyses

`Cm_Hawaii_HyPhy_Bayesiantree.fasta`: input to HyPhy tests

`Cm_Hawaii_IQTree.nwk`: output tree of IQTree run


